crates/cli/src/sync.rs

```diff
@@ -189,10 +189,18 @@ pub async fn config_sync(
             let encrypted_client_secret =
                 if let Some(client_secret) = provider.client_secret.as_deref() {
                     Some(encrypter.encrypt_to_string(client_secret.as_bytes())?)
-                } else if let Some(siwa) = provider.sign_in_with_apple.as_ref() {
-                    // For SIWA, we JSON-encode the config and encrypt it, reusing the client_secret
-                    // field in the database
-                    let encoded = serde_json::to_vec(siwa)?;
+                } else if let Some(mut siwa) = provider.sign_in_with_apple.clone() {
+                    // if private key file is defined and not private key (raw), we populate the
+                    // private key to hold the content of the private key file.
+                    // private key (raw) takes precedence so both can be defined
+                    // without issues
+                    if siwa.private_key.is_none() {
+                        if let Some(private_key_file) = siwa.private_key_file.take() {
+                            let key = tokio::fs::read_to_string(private_key_file).await?;
+                            siwa.private_key = Some(key);
+                        }
+                    }
+                    let encoded = serde_json::to_vec(&siwa)?;
```

Comment:
This does mean that it's possible to end up in the database with *no* private key if none are set. But I'm fine with that, SiWA is sufficiently niche anyway :)